process_model/process_model.py

```diff
@@ -1,72 +1,154 @@
-import click
-import re
+"""process_model outputs a Fortran module to interface to the given TF models
+
+process_model uses some of the methods provided in the TensorFlow
+saved_model_cli app to inspect a TensorFlow SavedModel and output the necessary
+Fortran to interface it to the fortran-tf-lib library.
+
+"""
+
 import sys
-from jinja2 import PackageLoader, Environment
-from typing import Dict, Optional, Tuple
+import re
+import click
+from typing import Any, Dict, List, Optional, Sequence, TextIO, Tuple
+from jinja2 import Environment, PackageLoader
 
 
-replacer = re.compile(r'^DT')
+replacer = re.compile(r"^DT")
 
 env = Environment(
     loader=PackageLoader(__name__),
     lstrip_blocks=True,
     trim_blocks=True,
 )
 
+# TODO: Consider adding type hints to the function defs.
+# TODO: Consider adding argument types and descriptions to docstrings.
+# TODO: consider linting with mypy, pydoctyle (numpy convention?) and pylint.
+
 
 @click.command()
-@click.argument('model_dirs', type=click.Path(exists=True,
-                file_okay=False, dir_okay=True), nargs=-1, required=True)
-@click.option('-o', '--output-file', default='-', type=click.File(mode='wt'),
-              help='Where to send Fortran output.')
-@click.option('-t', '--tag-set', default='serve', type=click.STRING,
-              help='tag-set to use (default `serve`).')
-@click.option('-m', '--module-name', default='ml_module', type=click.STRING,
-              help='Name to give the Fortran module (default `ml_module`).')
-@click.option('-s', '--signature-def', default='serving_default',
-              type=click.STRING, help='signature def to use (default '
-              '`serving_default`)')
-@click.option('--indent', default=4, type=click.INT,
-              help='Indentation level of output code (default 4)')
-def main(model_dirs, output_file, tag_set, module_name, signature_def, indent):
-    '''
+@click.argument(
+    "model_dirs",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+    nargs=-1,
+    required=True,
+)
+@click.option(
+    "-o",
+    "--output-file",
+    default="-",
+    type=click.File(mode="wt"),
+    help="Where to save the Fortran output (default stdout).",
+)
+@click.option(
+    "-t",
+    "--tag-set",
+    default="serve",
+    type=click.STRING,
+    help="tag-set to use (default `serve`).",
+)
+@click.option(
+    "-m",
+    "--module-name",
+    default="ml_module",
+    type=click.STRING,
+    help="Name to give the Fortran module (default `ml_module`).",
+)
+@click.option(
+    "-s",
+    "--signature-def",
+    default="serving_default",
+    type=click.STRING,
+    help="signature def to use (default " "`serving_default`)",
+)
+@click.option(
+    "--indent",
+    default=4,
+    type=click.INT,
+    help="Indentation level of output Fortran code (default 4)",
+)
+def main(
```

Comment:
Glorious!

Reply:
Not entirely sure `mypy` knows what to do with these but it doesn't complain.